deployment/katanaDeployment.js

```diff
@@ -0,0 +1,135 @@
+  parseInput(configParams);
+  const [deployer] = await ethers.getSigners();
+  deployerAddress = await deployer.getAddress();
+
+  // contract deployment
+  console.log('deploying staking contract...');
+  const KyberStaking = await ethers.getContractFactory('KyberStaking');
+  const kyberStaking = await KyberStaking.deploy(deployerAddress, kncAddress, epochPeriod, starttime);
+  await kyberStaking.deployed();
+  console.log(`staking address: ${kyberStaking.address}`);
+  await pressToContinue();
+
+  console.log('deploying governance contract...');
+  const KyberGovernance = await ethers.getContractFactory('KyberGovernance');
+  const kyberGovernance = await KyberGovernance.deploy(deployerAddress, daoOperator, [], []);
+  await kyberGovernance.deployed();
+  console.log(`governance address: ${kyberGovernance.address}`);
+  await pressToContinue();
+
+  console.log('deploying short executor...');
+  const Executor = await ethers.getContractFactory('DefaultExecutor');
+  const shortExecutor = await Executor.deploy(
+    kyberGovernance.address,
+    shortExecutorConfig.delay,
+    shortExecutorConfig.gracePeriod,
+    shortExecutorConfig.minimumDelay,
+    shortExecutorConfig.maximumDelay,
+    shortExecutorConfig.minVoteDuration,
+    shortExecutorConfig.maxVotingOptions,
+    shortExecutorConfig.voteDifferential,
+    shortExecutorConfig.minimumQuorum
+  );
+  await shortExecutor.deployed();
+  console.log(`shortExecutor address: ${shortExecutor.address}`);
+  await pressToContinue();
+
+  console.log('deploying long executor...');
+  const longExecutor = await Executor.deploy(
+    kyberGovernance.address,
+    longExecutorConfig.delay,
+    longExecutorConfig.gracePeriod,
+    longExecutorConfig.minimumDelay,
+    longExecutorConfig.maximumDelay,
+    longExecutorConfig.minVoteDuration,
+    longExecutorConfig.maxVotingOptions,
+    longExecutorConfig.voteDifferential,
+    longExecutorConfig.minimumQuorum
+  );
+  await longExecutor.deployed();
+  console.log(`longExecutor address: ${longExecutor.address}`);
+  await pressToContinue();
+
+  console.log('deploying voting power strat...');
+  const VotingPowerStrategy = await ethers.getContractFactory('EpochVotingPowerStrategy');
+  const votingPowerStrategy = await VotingPowerStrategy.deploy(
+    kyberGovernance.address,
+    kyberStaking.address
+  );
+  await votingPowerStrategy.deployed();
+  console.log(`votingPowerStrategy address: ${votingPowerStrategy.address}`);
+  await pressToContinue();
+
+  exportAddresses({
+      'staking': kyberStaking.address,
+      'governance': kyberGovernance.address,
+      'shortExecutor': shortExecutor.address,
+      'longExecutor': longExecutor.address,
+      'votingPowerStrategy': votingPowerStrategy.address
+  });
+
+  // set executors and voting power strategy in governance
+  console.log(`authorize executors...`);
+  await kyberGovernance.authorizeExecutors([shortExecutor.address, longExecutor.address]);
```

Comment:
This should be in constructor, right?

Reply:
circular dependency. executors need governance in ctor, governance needs executors in ctor.